assets/js/blocks/cart-checkout/checkout-i2/inner-blocks/checkout-terms-block/frontend.tsx

```diff
@@ -44,27 +45,22 @@ const FrontendBlock = ( {
 			return;
 		}
 		if ( checked ) {
-			clearValidationError( validationErrorId );
+			clearValidationError();
 		} else {
-			setValidationErrors( {
-				[ validationErrorId ]: {
-					message: __(
-						'Please read and accept the terms and conditions.',
-						'woo-gutenberg-products-block'
-					),
-					hidden: true,
-				},
+			setValidationError( {
+				message: __(
+					'Please read and accept the terms and conditions.',
+					'woo-gutenberg-products-block'
+				),
+				hidden: false,
 			} );
 		}
-		return () => {
-			clearValidationError( validationErrorId );
-		};
```

Comment:
This was the part that caused the infinite loop. I'm not sure why having it would change anything. Deleting it didn't break stuff.

Reply:
Looks like this was here to clear validation errors for a component if it were to be unmounted. A good test case would be for a validation error within the billing address section, and then hiding the billing address section?

Reply:
I will test it in that case

Reply:
This test case checks out. Will try to understand why is this loop happening.

Reply:
Fixed in 8dcc6ed367cb308fa8fd79ad87556d98676fb36e